docs: Add TextBox Placeholder and Header breaking change section (#1445)
unoplatform/Uno.Themes: doc/material-migration.md

The Uno Material v5 packages introduce a new dependency on the [Uno Themes](https://www.nuget.org/packages/Uno.Themes.WinUI) package. Uno Themes is the base library for all design system implementations going forward. As a result, the following changes have been made:

> ### TextBox `PlaceholderText` and `Header`

Reviewer: We should move this under a subsection with a title like Upgrading to 5.1 or something

Author: Changes pushed.

Author: ![image](https://github.com/user-attachments/assets/1ba72437-bc8f-484a-aa27-673cffa8fdfa)

The Uno Material v5 packages introduce a new dependency on the [Uno Themes](https://www.nuget.org/packages/Uno.Themes.WinUI) package. Uno Themes is the base library for all design system implementations going forward. As a result, the following changes have been made:

### Upgrading to Uno Themes 5.1

> #### TextBox `PlaceholderText` and `Header`

Reviewer: Is it only TextBox, or did we do this for PasswordBox too?

Reviewer 2: Hum, that is a really good point @kazo0!
@eriklimakc I went back to the main issue (https://github.com/unoplatform/Uno.Themes/issues/974) and it is mentioned that changes should normally be done the same for PasswordBox as well. 
But the related PR covers only TextBox: https://github.com/unoplatform/Uno.Themes/pull/1014

Reviewer 3: @agneszitte I think it would be very nice to add more examples for this, as it will give a good comparison between the options. Do you want me to take a look when this is merged?

Author: @agneszitte @kazo0 I thought we had agreed to do this only for the TextBox, but I can't find any message or comment confirming that. I might be confusing things. Apologies for missing it. 😞

If agreed by @kazo0, and @rajamatt could help with the PasswordBox that would be great. I can support as/if needed.

Thanks.

Author: Issue for the `PasswordBox` work: https://github.com/unoplatform/Uno.Themes/issues/1449

> Previously, the `PlaceholderText` attribute acted as a header, displaying in a normal state and animating upwards when text was entered. Now, both `PlaceholderText` and `Header` are supported and the behavior changed.

Reviewer 2: ```suggestion
Previously, the `PlaceholderText` attribute acted as a header, displaying in a normal state and animating upwards when the text was entered. Now, both `PlaceholderText` and `Header` are supported and the behavior changed.
```

> Previously, the `PlaceholderText` attribute acted as a header, displaying in a normal state and animating upwards when text was entered. Now, both `PlaceholderText` and `Header` are supported and the behavior changed.

Reviewer 4: ```suggestion
Previously, the `PlaceholderText` property acted as a header, displaying in the normal state and animating upwards when the text is set or being entered. And, the `Header` property did nothing. Now, both `PlaceholderText` and `Header` are supported and their behaviors have changed.
```

##### New Behavior

- **PlaceholderText**: Displays inside the TextBox when it's empty. Disappears when text is entered.

> - **Header**: Acts as a label. Animates upwards when the TextBox is clicked.

Reviewer 4: ```suggestion
- **Header**: Acts as a label. Animates upwards when the TextBox is focused.
```
